util/chplenv/chpl_jemalloc.py

```diff
@@ -1,26 +1,55 @@
 #!/usr/bin/env python3
 import os
 import sys
+import optparse
 
-import chpl_mem, overrides, third_party_utils
+import chpl_mem, chpl_platform, overrides, third_party_utils
 from utils import error, memoize, run_command, warning
 
 
 @memoize
-def get():
-    jemalloc_val = overrides.get('CHPL_JEMALLOC')
-    mem_val = chpl_mem.get('target')
-    if not jemalloc_val:
-        if mem_val == 'jemalloc':
+def get(target='target'):
+    chpl_jemalloc = overrides.get('CHPL_JEMALLOC')
+    chpl_host_jemalloc = overrides.get('CHPL_HOST_JEMALLOC')
+    chpl_target_jemalloc = overrides.get('CHPL_TARGET_JEMALLOC')
+    mem_val = chpl_mem.get(target)
+
+    platform_val = chpl_platform.get('target')
+    darwin = platform_val.startswith('darwin')
+    linux = platform_val.startswith('linux')
+
+    if target == 'target':
+        if chpl_target_jemalloc:
+            jemalloc_val = chpl_target_jemalloc
+        elif chpl_jemalloc:
+            jemalloc_val = chpl_jemalloc
+        else:
+            jemalloc_val = 'bundled'
+    else:
+        if chpl_host_jemalloc:
+            jemalloc_val = chpl_host_jemalloc
+        elif chpl_jemalloc:
+            jemalloc_val = chpl_jemalloc
+        elif linux and mem_val == 'jemalloc':
             jemalloc_val = 'bundled'
+        elif darwin and mem_val == 'jemalloc':
+            jemalloc_val = 'system'
         else:
             jemalloc_val = 'none'
 
+    # TODO what to do on crays?
```

Comment:
@ronawho forgot to bring this up, is there a preferred way to test for linux-y systems including cray?

Reply:
I think we tend to just assume if it's not mac or cygwin, then it's linux-y.